cmd/zetaclientd/utils.go

```diff
@@ -97,9 +80,68 @@ func waitForZetacoreToCreateBlocks(ctx context.Context, zc interfaces.ZetacoreCl
 			return fmt.Errorf("zetacore is not ready, timeout %s", time.Since(start).String())
 		}
 
-		logger.Info().Msgf("Failed to get block number, retry : %d/%d", retryCount, attempts)
-		time.Sleep(interval)
+		logger.Info().Msgf("Failed to get block number, retry: %d/%d", retryCount, attempts)
+		return nil
+	}
+
+	return ticker.Run(ctx, interval, task)
+}
+
+// prepareZetacoreClient prepares the zetacore client for use.
+// EXITS THE PROGRAM IF THIS NODE IS NOT AN OBSERVER.
+func prepareZetacoreClient(ctx context.Context, zc *zetacore.Client, cfg *config.Config, logger zerolog.Logger) error {
+	// Set grantee account number and sequence number
+	if err := zc.SetAccountNumber(authz2.ZetaClientGranteeKey); err != nil {
+		return errors.Wrap(err, "failed to set account number")
+	}
+
+	res, err := zc.GetNodeInfo(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to get node info")
+	}
+
+	network := res.GetDefaultNodeInfo().Network
+	if network != cfg.ChainID {
+		logger.Warn().
+			Str("got", cfg.ChainID).
+			Str("want", network).
+			Msg("Zetacore chain id config mismatch. Forcing update from the network")
+
+		cfg.ChainID = network
+		if err = zc.UpdateChainID(cfg.ChainID); err != nil {
+			return errors.Wrap(err, "failed to update chain id")
+		}
+	}
+
+	isObserver, err := isObserverNode(ctx, zc)
```

Comment:
Wont this check new nodes from taking part in the TSS ceremony .
Conisder the situation when a new node is added , and is part of the node account list but not part of the observer set . 
They need to sign on the TSS ceremony.

I think this check is correct , but it should be done after the TSS voting is executed

Reply:
Okay, initially it was added here
https://github.com/zeta-chain/node/pull/1031/files#diff-4f6dad4b36c4f2c7c015aaa31c4468e0fd88293d9fbf4a5f9c1be1751ea5ebe6R184

I just refactored the semantics. Will move below, thanks.